src/pages/index.js

```diff
@@ -21,91 +34,155 @@ popUpImageModal.setEventListeners();
 editFormValidator.enableValidation();
+//avatar Form - link
+const avatarFormValidator = new FormValidator(
+  constants.settings,
+  constants.avatarForm
+);
+avatarFormValidator.enableValidation();
+//card form place/Url
 const addFormValidator = new FormValidator(
   constants.settings,
   constants.addModalForm
 );
 addFormValidator.enableValidation();
-
-=======
-avatarFormValidator.enableValidation();
->>>>>>> Stashed changes
 // call queryselector on input inside form /set values
 function handleFormFill(userInfoList) {
   constants.profileTitleInput.value = userInfoList.name;
   constants.profileDescriptionInput.value = userInfoList.description;
 }
-//Buttons that Open Popup with forms
-const newUserInfo = new UserInfo(constants.selectors);
-constants.profileEditButton.addEventListener("click", () => {
-  // grab from userinfo name + descriptions
-  handleFormFill(newUserInfo.getUserInfo());
 
-  return profilePopup.openModal();
-});
 const profilePopup = new PopupWithForm(
   constants.selectors.profileSelector,
   handleProfileEditSubmit
 );
 profilePopup.setEventListeners();
+
 const cardPopUp = new PopupWithForm(
   constants.selectors.addSelector,
   handleAddModalSubmit
 );
 cardPopUp.setEventListeners();
+
+const avatarPopUp = new PopupWithForm(
+  constants.selectors.avatarSelector,
+  handleAvatarSubmit
+);
+avatarPopUp.setEventListeners();
+//Buttons that Open Popup with forms
+// grab from userinfo name + descriptions
+const newUserInfo = new UserInfo(constants.selectors);
+constants.profileEditButton.addEventListener("click", () => {
+  handleFormFill(newUserInfo.getUserInfo());
+  return profilePopup.openModal();
+});
 constants.addButton.addEventListener("click", () => {
-  addFormValidator.toggleButtonState();
   return cardPopUp.openModal();
 });
+
+constants.profileAvatar.addEventListener("click", () => {
+  return avatarPopUp.openModal();
+});
+
 //initializes new section renders inittial cards and new ones
-const cardSection = new Section(
-  {
-    items: constants.initialCards,
-    renderer: renderCard,
-  },
-  constants.selectors.cardSection
-);
-cardSection.renderItems(constants.initialCards);
+let cardSection;
+api.getInitialCards().then((data) => {
+  cardSection = new Section(
+    {
+      items: data,
+      renderer: renderCard,
+    },
+    constants.selectors.cardSection
+  );
+  cardSection.renderItems();
+});
+
 //Submit Button handler
+//Toggle Submit button State
 /**Event Handlers*/
-function handleProfileEditSubmit(modalInputs) {
-  newUserInfo.setUserInfo(modalInputs.title, modalInputs.description);
-  profilePopup.closeModal();
+function handleProfileEditSubmit(modalInputs, popUpForm) {
+  api
+    .updateInfo(modalInputs)
+    .then(() => {
+      const { title, description } = modalInputs;
+      newUserInfo.setUserInfo(title, description);
+      editFormValidator.toggleButtonState();
+    })
+    .finally(() => {
+      popUpForm.querySelector(".modal__form-button").innerText = "Save";
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```